### PR TITLE
Render models with the new datamodel renderer

### PR DESCRIPTION
**Pull request by pimeys**
Part of: https://github.com/prisma/prisma/issues/15800

Some of the rendering tests we should maybe redo in introspection. Let's see about that in the upcoming PRs. Lot of the code from here we can remove when we remove DML from introspection.

### CODE REVIEW

File: psl/psl/tests/attributes/index_clustering.rs [hunk 1]
```diff
@@ -120,248 +120,3 @@ fn non_clustered_compound_id_works_on_sql_server() {
         clustered: Some(false),
     });
 }
-
-#[test]
-fn do_not_render_id_default_clustering() {
```
Comment (pimeys) [inline psl/psl/tests/attributes/index_clustering.rs:125]: These are all tested in intro.

File: psl/psl/tests/attributes/index_positive.rs [hunk 2]
```diff
@@ -163,22 +162,6 @@ fn multiple_index_must_work() {
     });
 }
 
-#[test]
-fn index_attributes_must_serialize_to_valid_dml() {
```
Comment (pimeys) [inline psl/psl/tests/attributes/index_positive.rs:167]: We test these extensively in intro.

File: psl/psl/tests/attributes/relations/relations_positive.rs [hunk 3]
```diff
@@ -249,222 +248,6 @@ fn allow_explicit_fk_name_definition() {
         .assert_relation_fk_name(Some("CustomFKName".to_string()));
 }
 
-#[test]
-fn allow_implicit_fk_name_definition() {
```
Comment (pimeys) [inline psl/psl/tests/attributes/relations/relations_positive.rs:253]: The DML has a wiped constraint name for default values.

We test these in the ME.

File: psl/psl/tests/attributes/unique_positive.rs [hunk 4]
```diff
@@ -281,22 +280,6 @@ fn single_field_unique_indexes_on_enum_fields_must_work() {
     });
 }
 
-#[test]
-fn unique_attributes_must_serialize_to_valid_dml() {
```
Comment (pimeys) [inline psl/psl/tests/attributes/unique_positive.rs:285]: Tested in intro.

File: psl/psl/tests/base/unsupported_type.rs [hunk 5]
```diff
@@ -1,45 +0,0 @@
-use crate::common::*;
-
-#[test]
-fn parse_unsupported_types() {
```
Comment (pimeys) [inline psl/psl/tests/base/unsupported_type.rs:4]: Parsing tested in many places in PSL. Rendering tested in intro.

File: psl/psl/tests/renderer/extended_indexes.rs [hunk 6]
```diff
@@ -1,116 +0,0 @@
-use crate::{common::*, with_header, Provider};
```
Comment (pimeys) [inline psl/psl/tests/renderer/extended_indexes.rs:1]: All tested in intro/migrations.

File: psl/psl/tests/renderer/literals.rs [hunk 7]
```diff
@@ -1,159 +0,0 @@
-use crate::common::*;
-use psl::dml::{DefaultValue, PrismaValue};
-
-#[test]
-fn strings_with_quotes_render_as_escaped_literals() {
```
Comment (pimeys) [inline psl/psl/tests/renderer/literals.rs:5]: Added one test for this to intro. Migrations are already testing these.

File: psl/psl/tests/renderer/simplification.rs [hunk 8]
```diff
@@ -1,91 +0,0 @@
-use crate::common::*;
-
-#[test]
-fn test_must_not_render_relation_fields_with_many_to_many() {
```
Comment (pimeys) [inline psl/psl/tests/renderer/simplification.rs:4]: Formatter tests?

File: psl/psl/tests/renderer/simplification.rs [hunk 9]
```diff
@@ -1,91 +0,0 @@
...
-          Post Post[] @relation(references: [id])
-        }
-    "#]];
-
-    let result = psl::render_datamodel_to_string(&parse(input), None);
-    expected.assert_eq(&result);
-}
-
-#[test]
-fn test_exclude_default_relation_names_from_rendering() {
```
Comment (pimeys) [inline psl/psl/tests/renderer/simplification.rs:34]: Formatter?

**Review by tomhoule — APPROVED** [review]
Reviewed in a zoom call with @eviefp and @pimeys 

:chefkiss:

**Pull request merged**